### ISSUE TITLE
WSL: "--distribution" flag breaks cargo commands

### ISSUE THREAD

**Issue by secana** [intellij-rust/intellij-rust#9061]
## Environment

* **IntelliJ Rust plugin version: 0.4.174.4743-221**
* **Rust toolchain version: 1.62.0**
* **IDE name and version: CLion 2022.1.3**
* **Operating system: Windows 11 + WSL Manjaro**

## Problem description

When running **CLion** with the **intellij-rust** and **WSL** as the target, the flag `--distribution Manjaro` is added each `cargo` command. Unfortunately, this flag does not exists, such that all command fail and running or testing Rust code doesn't work.

Below the output, that is printed by CLion:

```bash
/home/x/.cargo/bin/cargo --distribution Manjaro --exec /bin/sh -c export PATH="/home/x/.cargo/bin:$PATH" && export RUST_BACKTRACE=short && export TERM=ansi && cd /home/x/repos/bitfalter/kellnr && /home/x/.cargo/bin/cargo check --color=always
error: Found argument '--distribution' which wasn't expected, or isn't valid in this context

	If you tried to supply `--distribution` as a value rather than a flag, use `-- --distribution`

USAGE:
    cargo [+toolchain] [OPTIONS] [SUBCOMMAND]

For more information try --help

Process finished with exit code 1
```

## Steps to reproduce

Target configuration:

![image](https://user-images.githubusercontent.com/12230822/178944236-55e45fa5-74a5-441c-bffc-3e00c8cb7ae1.png)

Run configuration:

![image](https://user-images.githubusercontent.com/12230822/178944434-a3e1a0e3-4229-4d63-a3ad-990fb835e12d.png)

Run any cargo command through the IDE and it will fail with the error above.

**Comment by secana** [comment]
The issue seems not to occur, if the target is set as "default" and then the "default" is used, but I guess it should work for all targets, not only the default one.

**Comment by PeterGrace** [comment]
I think that this first command that is executed should be wsl.exe (which accepts the distribution flag) as its attempting to run shell commands further down the command pipeline, if you scroll over.  I'm also having this problem with CLion 2022.1, windows 10, wsl2 (ubuntu.)  I think there's probably a place in the code that is supposed to be calling wsl but somehow got transposed to cargo, that, or there's a settings option somewhere that is ambiguously named that really means "which command to execute to make calls into wsl."

**Comment by ejohb** [comment]
So is WSL debugging just completely broken at present?

**Comment by vlad20012** [comment]
@ejohb it seems broken using `Run Targets`, but it works if you set up a [WSL toolchain](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-project-settings.html#wsl-toolchain)

**Comment by lilianmoraru** [comment]
![image](https://user-images.githubusercontent.com/621738/197576407-7696a5df-9c2d-4f7c-9fd3-3fd01845ee7d.png)

@vlad20012 I have a WSL toolchain and it still passes the `--distribution` flag, which leads to the error:
```bash
/home/lilian/.cargo/bin/cargo --distribution Ubuntu-20.04 --exec /bin/sh -c export PATH="/home/lilian/.cargo/bin:$PATH" && export CARGO_TERM_PROGRESS_WIDTH=1000 && export RUST_BACKTRACE=short && export TERM=ansi && export CARGO_TERM_PROGRESS_WHEN=always && cd /home/lilian/<rust-project> && /home/lilian/.cargo/bin/cargo build --color=always --message-format=json-diagnostic-rendered-ansi --verbose

error: Found argument '--distribution' which wasn't expected, or isn't valid in this context
```

**Comment by vlad20012** [comment]
@lilianmoraru see [this answer](https://github.com/intellij-rust/intellij-rust/issues/9467#issuecomment-1268544960).

Your screenshot is about C++ toolchain, not Rust toolchain

**Comment by lilianmoraru** [comment]
@vlad20012 I initially used this UI to set it up but it was failing due to the `--distribution` flag:
![image](https://user-images.githubusercontent.com/621738/197578118-cd34d5d8-31ac-4bbc-b9f9-4de63160a39e.png)

Ok, I see it now. It's the same naming and I thought it picked the C++ toolchain.
For people stumbling on this: I guess you need to create a new Rust toolchain("Manage targets...") and then either create a new config(chooses "Project default" and has the WSL icon) or manually switch back to "Project default":
![image](https://user-images.githubusercontent.com/621738/197579401-13684f31-7a64-4aa7-aa89-39c03ee233b5.png)

**Comment by Anivie** [comment]
This issue still exists in RustRover, how can I fix it?
![image](https://github.com/intellij-rust/intellij-rust/assets/88354708/15cf3e82-f2c8-4eca-8efd-36d63190f6f3)

**Comment by andreamancuso** [comment]
Thank you @Anivie that fixed it for me. How I wish this were more intuitive.
